setup.py

```diff
@@ -4,15 +4,13 @@
 from setuptools import setup, find_packages, Command
 
 __version__ = None  # Overwritten by executing version.py.
-with open('puncover/version.py') as f:
-    exec (f.read())
-
-with open('requirements.txt') as f:
-    requires = f.readlines()
```

Comment:
what's the benefit of removing `requirements.txt`? I found it useful to have it for local development and then referring to it through this `setup.py` to stay DRY. Is this in conflict with tax?

Reply:
Also, it seems that by removing `codecov` here, the top-level command line tool is missing and travis won't upload test coverage any longer:
```
$ codecov
codecov: command not found
```

We may want to add that dependency explicitly to the `.travis.yml` now – it was technically never a runtime requirement (but development requirement only) anyway.

Reply:
Yeah, there are a few quirky things with the `test_requires`, `requires`, and `extra_requires` we'll work out to make things cleaner. 

You don't need a `requirements.txt` to stay dry, users will just `pip install .`, `pip install -e .` or `pip install git+...` and they'll have everything they need installed. No need for a `requirements.txt`.